### ISSUE TITLE
ham/spam learning issue

### ISSUE THREAD

**Issue by andrewsav-datacom** [hardware/mailserver#110]
As far as I can see in the image ham is learned on a daily schedule. What if users are not watching their mailboxes every second and did not remove some spam from inbox before the learning kicks it. Will spam be learned as ham?

**Comment by hardware** [comment]
ham/spam learning will be improved (realtime learn) with Rspamd and Dovecot Antispam plugin in the next major release.

**Comment by hardware** [comment]
Issue related : #76

**Comment by michael-k** [comment]
Release date for [stretch](https://www.debian.org/releases/stretch/) is [2017-06-17](https://lists.debian.org/debian-devel-announce/2017/05/msg00002.html) :)

**Comment by hardware** [comment]
**Fixed in 1.1 (beta)**

Rspamd automatically learns when a false negative is moved to the Spam folder or when a false positive is moved out of the Spam folder.

|source | destination | action |
|--|--|--|
|* | Spam | report-spam.sieve
|Spam | * | report-ham.sieve

```
# https://wiki2.dovecot.org/HowTo/AntispamWithSieve

imapsieve_mailbox1_name = Spam
imapsieve_mailbox1_causes = COPY
imapsieve_mailbox1_before = file:/etc/dovecot/sieve/report-spam.sieve

imapsieve_mailbox2_name = *
imapsieve_mailbox2_from = Spam
imapsieve_mailbox2_causes = COPY
imapsieve_mailbox2_before = file:/etc/dovecot/sieve/report-ham.sieve
```


Rspamd WebUI shows the number of message learned :

![screenshot_20170618_155843](https://user-images.githubusercontent.com/327354/27261227-1caf08dc-543f-11e7-8ba0-8d3f8afa513f.png)

```bash
# docker logs -f mailserver

dovecot: imap(contact@domain.tld): sieve: pipe action: piped message to program 'rspamd-pipe-spam.sh'
dovecot: imap(contact@domain.tld): sieve: pipe action: piped message to program 'rspamd-pipe-ham.sh'
```
